### ISSUE TITLE
GlobalShortcut_unix: evdev mode should not fall back to XInput2 when no keyboard device nodes area available

### ISSUE THREAD

**Issue by mkrautz** [mumble-voip/mumble#3269]
Via https://github.com/mumble-voip/mumble/issues/3243#issuecomment-347015908

Evdev mode should not fall back to XInput2 when no keyboard device nodes are available:

> However, there is a bit of a misbehavior right now, where Mumble will fall back to XInput2 when no keyboards can be opened via evdev. This behavior is from back when evdev was our default.
Now, it makes more sense for Mumble to keep using evdev if it is enabled, and warn the user if no keyboard device nodes are available.

Instead, we should just warn the user, and keep on chugging along, allowing users to use evdev mode to allow global shortcuts for mice, for example.

**Comment by mkrautz** [comment]
Perhaps we should warn for each of
- Keyboard
- Mouse
- Joystick
?

**Comment by the-sane** [comment]
Yes, since the user has to specifically enable evdev, mumble should assume they know what they're doing and not prevent them from doing it.

In my case, this prevents me from assigning PTT to a joystick button because mumble falls back to XInput.
I assume there's no way for me to force evdev until this is resolved?

**Comment by Krzmbrzl** [comment]
> I assume there's no way for me to force evdev until this is resolved?

I don't think so :thinking: 
@davidebeatrici might know more though

**Comment by davidebeatrici** [comment]
I confirm that unfortunately there is no way to force evdev:
https://github.com/mumble-voip/mumble/blob/919f1f7a3f489a1d43475b07b6ae35196889c474/src/mumble/GlobalShortcut_unix.cpp#L67-L86

I will take care of it.
